Adds docs for deploying signoz to OpenShift (#1567)
SigNoz/signoz.io: data/docs/install/kubernetes/openshift.mdx

---
date: 2025-06-17
id: openshift
title: Deploying to OpenShift

> description: Instructions to install SigNoz on Openshift cluster

Reviewer: ```suggestion
description: Instructions to install SigNoz on OpenShift cluster
```

---

import CommonPrerequisites from '@/components/shared/k8s-common-prerequisites.md'
import K8sInstall from '@/components/shared/k8s-install.mdx'
import K8sOtelDemo from '@/components/shared/k8s-oteldemo.mdx'
import K8sNextSteps from '@/components/shared/k8s-next-steps.md'
import Admonition from '@/components/Admonition/Admonition'

> <HostingDecision />

Reviewer: ```suggestion
```

Not sure if this helps the user in any way.

## Prerequisites

> - Set up a OpenShift cluster - OCP/ROSA/CRC (see the [official OpenShift documentation][1] for more info).

Reviewer: ```suggestion
- Existing [OpenShift cluster](https://docs.redhat.com/en/documentation/red_hat_advanced_cluster_management_for_kubernetes/2.2/html/manage_cluster/creating-a-cluster) - OCP/ROSA/CRC (see the [official OpenShift documentation][1] for more info).
```

- `oc` cli configured

> <CommonPrerequisites />

Reviewer: <img width="510" alt="Screenshot 2025-06-17 at 8 18 39 PM" src="https://github.com/user-attachments/assets/5cb464ac-35a3-4893-9374-bfdb410bd85b" />
Make this table centre-aligned with proper spacing.

Also, does the user need to have both oc and kubectl configured? If not, we should combine them in a single pointer.

Author: Sure, copied the table from the shared component - will also update the shared component in the next PR

> 

Reviewer: ## Setup

### Step 1:
### Step 2:
### Step 3 (Optional):
### Step 4 (Optional):

> ## Configuring the Security Context Constraints (SCC)

Reviewer: ```suggestion
## Configure Security Context Constraints (SCC)
```

> OpenShift enforces SecurityContextConstraints (SCCs) to control what actions pods can perform and what resources they can access—similar to how RBAC controls user permissions. SCCs let administrators define the security conditions a pod must meet to be admitted and run in the cluster (see the [official Openshift documentation][4] for more info)

Reviewer: ```suggestion
OpenShift enforces [SecurityContextConstraints](https://docs.redhat.com/en/documentation/openshift_container_platform/3.11/html/cluster_administration/admin-guide-manage-scc) (SCCs) to control what actions pods can perform and what resources they can access, similar to how RBAC controls user permissions. SCCs let administrators define the security conditions a pod must meet to be admitted and run in the cluster.
```

We need to define the following SCC for running signoz on OpenShift cluster:

```yaml
apiVersion: security.openshift.io/v1
kind: SecurityContextConstraints
metadata:
  name: signoz-scc
allowPrivilegedContainer: false
runAsUser:
  type: RunAsAny
fsGroup:
  type: RunAsAny
seLinuxContext:
  type: RunAsAny
volumes:
  - persistentVolumeClaim
  - emptyDir               
  - configMap        
  - projected         
  - downwardAPI          
  - hostPath
readOnlyRootFilesystem: false
allowHostDirVolumePlugin: true
users:
  # These are the Service Accounts which are created during the Installation
  - system:serviceaccount:<NAMESPACE>:signoz-clickhouse

>   - system:serviceaccount:<NAMESPACE:default

Reviewer: ```suggestion
  - system:serviceaccount:<NAMESPACE>:default
```

>   - system:serviceaccount:<NAMESPACE:default

Reviewer: Explain in a couple what this codeblock will do.

```

Use the Kubectl to add the SCCs to OC cluster
```bash
kubectl apply -f signoz_scc.yaml
```
Or
Use the `oc` cli 
```bash

> oc apply -f signoz_scc.yaml

Reviewer: Convert these into tabs with lables **oc CLI (Recommended)** and **Kubectl**

</Admonition>

3. Install SigNoz:

```bash
helm repo add signoz https://charts.signoz.io
helm repo update
helm install signoz signoz/signoz \
   --namespace <namespace> --create-namespace \
   --wait \
   --timeout 1h \
   -f values.yaml
```

### Test the installation

1. In another terminal, port-forward signoz on its http port. (By default, signoz exposes its http server on port 8080.)

```bash
kubectl port-forward -n <namespace> svc/signoz 8080:8080
```

2. Run the following command to check the health of signoz:

```bash
curl -X GET http://localhost:8080/api/v1/health
```

3. If the installation is successful, you should see the following output:

```bash
{"status":"ok"}
```

<Admonition type="info">
By default, retention period is set to **7 days** for logs and traces, and **30 days** for metrics.
To change this, navigate to the **General** tab on the **Settings** page of SigNoz UI.

For more details, refer to https://signoz.io/docs/userguide/retention-period.
</Admonition>

## (Optional) Install OpenTelemetry Demo

<K8sOtelDemo />

## (Optional) SCCs for running K8s-Infra (Kubernetes Monitroing)

> We require following SCCs to be added the openShift cluster for the installation of K8s-infra for Kubernetes Monitroing

Reviewer: ```suggestion
We require following SCCs to be added the OpenShift cluster for the installation of K8s-infra for Kubernetes Monitoring
```

```yaml
apiVersion: security.openshift.io/v1
kind: SecurityContextConstraints
metadata:
  name: signoz-k8s-infra-agent-scc
allowPrivilegedContainer: false
allowHostPorts: true
runAsUser:
  type: RunAsAny
fsGroup:
  type: RunAsAny
seLinuxContext:
  type: RunAsAny
volumes:
  - persistentVolumeClaim
  - configMap              
  - projected          
  - downwardAPI      
  - hostPath
  - secret
readOnlyRootFilesystem: false
allowHostDirVolumePlugin: true
# <— bind this SCC only to the k8s-infra-otel-agent in the namespace
users:
  - system:serviceaccount:<NAMESPACE>:k8s-infra-otel-agent
```

## Next Steps

> <K8sNextSteps />

Reviewer: ```suggestion
- [Kubernetes Infra Monitoring](https://signoz.io/docs/userguide/k8s-metrics/)
- [OpenTelemetry Operator for automatic instrumentation](https://signoz.io/docs/tutorial/opentelemetry-operator-usage/#opentelemetry-auto-instrumentation-injection)
```

---

[1]: https://docs.redhat.com/en/documentation/red_hat_advanced_cluster_management_for_kubernetes/2.2/html/manage_cluster/creating-a-cluster
[2]: https://github.com/SigNoz/charts/tree/main/charts/signoz#configuration
[3]: https://docs.redhat.com/en/documentation/red_hat_openshift_container_storage/4.6/html-single/managing_and_allocating_storage_resources/index

> [4]: https://docs.redhat.com/en/documentation/openshift_container_platform/3.11/html/cluster_administration/admin-guide-manage-scc

Reviewer: ```suggestion
```

DIrectly hyperlinked.

We need to define the following SCC for running SigNoz on OpenShift cluster:

```yaml
apiVersion: security.openshift.io/v1
kind: SecurityContextConstraints
metadata:
  name: signoz-scc

allowPrivilegedContainer: false
runAsUser:
  type: RunAsAny        # allow any container UID
fsGroup:
  type: RunAsAny        # allow any supplemental GID
seLinuxContext:
  type: RunAsAny        # no SELinux label constraints
volumes:
  - persistentVolumeClaim # mounted PVC
  - emptyDir              # ephemeral scratch space
  - configMap             # ConfigMap files
  - projected             # combined CM/secret/etc.
  - downwardAPI           # expose pod metadata

readOnlyRootFilesystem: false
allowHostDirVolumePlugin: false 

users:
  # Bind the SCC to Required Service Accounts for SigNoz
  - system:serviceaccount:<NAMESPACE>:signoz-clickhouse
  - system:serviceaccount:<NAMESPACE>:default

> ```

Reviewer: Please explain this code block in a couple of lines.

Author: Added brief explanation for the SCCs
